gcloud-java-storage/src/main/java/com/google/gcloud/storage/BlobReadChannel.java

```diff
@@ -46,4 +45,27 @@
    */
   void chunkSize(int chunkSize);
 
+  /**
+   * Saves the read channel state.
+   *
+   * @return an object that contains the read channel state and can restore it afterwards. State
+   *     object must implement {@link java.io.Serializable}.
+   */
+  public State save();
+
+  /**
+   * A common interface for all classes that implement the internal state of a
+   * {@code BlobReadChannel}.
+   *
+   * Implementations of this class must implement {@link java.io.Serializable} to ensure that the
+   * state of a channel can be correctly serialized.
+   */
+  public interface State {
```

Comment:
Yes we can make it a generic interface in `core` even though I don't know how much it will be reused. If we make it top level we probably should change the name (`State` is too vague) to something more specific, `RestorableState` maybe?